### PR TITLE
feature(issue): include most recent rotated log backup in archive

### PR DESCRIPTION
**Pull request by garmr-ulfr**
This pull request enhances the log archiving process by ensuring that the most recent rotated (gzip-compressed) log backup is included at the start of the main log in the issue archive, preserving more complete log history across log rotations. It also introduces robust helper functions and comprehensive tests for this new behavior.

**Log backup inclusion and ordering:**
- The `buildIssueArchive` function now prepends the most recent compressed backup (if present) to the primary log, ensuring that log history from before a rotation is not lost. The backup is trimmed to fit the archive size limit, and a newline is inserted if the backup doesn't end with one.

**Helper functions for backup handling:**
- Added `prependMostRecentBackup`, `findMostRecentCompressedBackup`, and `readGzipTail` to locate the latest compressed backup, read its decompressed tail efficiently, and combine it with the current log.
- Introduced `maxLogReadFactor` constant to consistently cap how much uncompressed log data is read, replacing magic numbers. [[1]](diffhunk://#diff-e1747d137e525bb2c011c3f6a5968caeb5dfbad026c27efd490351250a618f50R6-R19) [[2]](diffhunk://#diff-e1747d137e525bb2c011c3f6a5968caeb5dfbad026c27efd490351250a618f50L77-R175)

**Testing improvements:**
- Added extensive tests for backup detection, reading, and integration into the archive, including edge cases like corrupt or empty backups, and correct ordering of backup and current log data. (F5b814d6L450R450)
- Refactored tests to use a new `randomBytes` helper for generating incompressible random data, improving clarity and consistency. [[1]](diffhunk://#diff-213de489f65fd3e4ac6bb678f3b59d4411bf1a03217c3fd858b5e6de813b3bb4L238-R239) [[2]](diffhunk://#diff-213de489f65fd3e4ac6bb678f3b59d4411bf1a03217c3fd858b5e6de813b3bb4L278-R277) [[3]](diffhunk://#diff-213de489f65fd3e4ac6bb678f3b59d4411bf1a03217c3fd858b5e6de813b3bb4L396-R394) [[4]](diffhunk://#diff-213de489f65fd3e4ac6bb678f3b59d4411bf1a03217c3fd858b5e6de813b3bb4L410-R405) F5b814d6L450R450)

**Test utilities:**
- Added `writeGzipFile` and `randomBytes` helpers to simplify test setup for compressed log files and random data generation. (F5b814d6L450R450)

**Dependency updates:**
- Imported the standard library `compress/gzip` package in both implementation and test files to support reading and writing gzip-compressed logs. [[1]](diffhunk://#diff-e1747d137e525bb2c011c3f6a5968caeb5dfbad026c27efd490351250a618f50R6-R19) [[2]](diffhunk://#diff-213de489f65fd3e4ac6bb678f3b59d4411bf1a03217c3fd858b5e6de813b3bb4R6)

## Summary by CodeRabbit

* **Bug Fixes**
  * Issue archives now include the latest available rotated compressed log backup, giving more complete log bundles for troubleshooting.
  * Archive generation is more reliable when backups are missing, empty, or corrupted, and preserves proper log ordering and line breaks.

* **Tests**
  * Expanded automated coverage for archive creation and compressed backup handling.

### CODE REVIEW

File: issue/archive.go [hunk 1]
```diff
@@ -42,9 +49,100 @@ func buildIssueArchive(logDir string, additionalFiles []string, maxSize int64) (
...
+	matches, err := filepath.Glob(pattern)
+	if err != nil || len(matches) == 0 {
+		return "", false
+	}
+
+	latest := matches[0]
+	for _, match := range matches[1:] {
+		if match > latest {
+			latest = match
+		}
```
[the remark this answers is not in the document]
Comment (garmr-ulfr) [reply issue/archive.go]: Fixed. It now attempts to parse the timestamp from the filename, which is formatted as `<log-name>-<timestamp>.log.gz`, by trimming `log-name` and `.log.gz` and skipping any that fail.

File: issue/archive.go [hunk 2]
```diff
@@ -42,9 +49,100 @@ func buildIssueArchive(logDir string, additionalFiles []string, maxSize int64) (
...
+	return latest, true
+}
+
+// readGzipTail returns up to maxRead bytes from the end of the decompressed file.
+// The tail is kept rather than the head because it is the history nearest in time
+// to the current log; memory stays bounded to ~maxRead even when the backup
+// decompresses to more.
+func readGzipTail(path string, maxRead int64) ([]byte, error) {
+	f, err := os.Open(path)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (garmr-ulfr) [reply issue/archive.go:141]: Added check.

File: issue/archive.go [hunk 3]
```diff
@@ -42,9 +49,100 @@ func buildIssueArchive(logDir string, additionalFiles []string, maxSize int64) (
...
+// prependMostRecentBackup prepends the newest rotated gzip backup, if present,
+// so a mid-session rotation does not lose earlier log history.
+func prependMostRecentBackup(primaryLogPath string, current []byte, maxArchiveSize int64) []byte {
+	backupPath, ok := findMostRecentCompressedBackup(primaryLogPath)
+	if !ok {
+		return current
+	}
+
+	maxRead := maxArchiveSize * maxLogReadFactor
+	backupData, err := readGzipTail(backupPath, maxRead)
```
[the remark this answers is not in the document]
Comment (garmr-ulfr) [reply issue/archive.go]: Done.

File: issue/archive.go [hunk 4]
```diff
@@ -3,13 +3,20 @@ package issue
...
 	"io"
 	"log/slog"
 	"os"
 	"path/filepath"
+	"strings"
 )
 
+// maxLogReadFactor bounds how much uncompressed log to read toward an archive:
+// maxArchiveSize * maxLogReadFactor bytes. Even with poor compression, the
+// archive can't need more than this to reach its size budget.
```
[the remark this answers is not in the document]
Comment (garmr-ulfr) [reply issue/archive.go]: Corrected comment.
